src/main/java/com/fortify/plugin/jenkins/steps/FortifyStep.java

```diff
@@ -54,47 +54,61 @@ protected void setLastBuild(Run<?, ?> lastBuild) {
 	}
 
 	/**
-	 * Search for the executable filename in FORTIFY_HOME or PATH environment
-	 * variables or workspace
-	 * 
+	 * Search for the executable filename in executable home directory or on PATH environment
+	 * variable or in workspace
+	 *
 	 * @param filename
-	 * @param checkFortifyHome
 	 * @param build
 	 * @param workspace
-	 * @param launcher
 	 * @param listener
-	 * @param msg
+	 * @param targetEnvVarName
 	 * @return found executable
 	 * @throws InterruptedException
 	 * @throws IOException
 	 */
-	protected String getExecutable(String filename, boolean checkFortifyHome, Run<?, ?> build, FilePath workspace,
-			Launcher launcher, TaskListener listener, String msg) throws InterruptedException, IOException {
-		PrintStream log = listener.getLogger();
+	protected String getExecutable(String filename, Run<?, ?> build, FilePath workspace,
+								   TaskListener listener, String targetEnvVarName) throws InterruptedException, IOException {
+		PrintStream logger = listener.getLogger();
 		EnvVars env = build.getEnvironment(listener);
-		String fortifyHome = null;
+
+		String home = null;
 		String path = null;
+		boolean isEnvVarSetProperly = true;
 		// check env variables defined in Jenkins master
 		for (Map.Entry<String, String> entry : env.entrySet()) {
-			String key = entry.getKey();
-			if ("FORTIFY_HOME".equals(key)) {
-				if (checkFortifyHome) {
-					fortifyHome = entry.getValue();
-					if (fortifyHome.endsWith("bin") || fortifyHome.endsWith("bin/") || fortifyHome.endsWith("bin\\")) {
-						log.println("WARNING: Environment variable FORTIFY_HOME should not point to bin directory");
-					}
+			String envVarName = entry.getKey();
+			String envVarValue = entry.getValue();
+			if (targetEnvVarName != null && targetEnvVarName.equals(envVarName)) {
```

Comment:
As far as I can see the logic of the cycle is designed to set 
1) home variable properly
2) path variable properly

But the logic might be broken in future if two conditions are met:
1) somebody invokes getExecutable() for "PATH" env variable
2) env.entrySet() will return "PATH" as first object.
This case the "else if("PATH"...)" condition will be never reached and isEnvVarSetProperly will be assigned to false mostly (because most of Path values end with bin).
I believe if the method is designed to set path variable despite value of targetEnvVarName, it should be changed to guarantee the path is set properly all the time.

Reply:
Thank you for your comment.

That's true that method `getExecutable` iterates all env variables and sets home variable and/or path variable. But it does not need to set them both as we only need one variable with a value of a location for finding executable. They also should be assigned to proper variables as the following after it logic is based on that.

Initially, this method was designed in a way that it prioritizes any user defined variable to use for finding executable and if such variables do not exist, then it gives next priority to FORTIFY_HOME env variable, and if the latter does not exist then it is looking within the PATH.  You can check [this comment](https://github.com/jenkinsci/fortify-plugin/issues/13#issuecomment-597368068) and [this code](https://github.com/jenkinsci/fortify-plugin/blob/6b2f9cb488d173accc1fbe01e077e4f888640d58/src/main/java/com/fortify/plugin/jenkins/steps/FortifyStep.java#L76-L85) to make sure. I am trying to say that we should not change the order of conditions.

Your 2 points are correct too, as bug in 1st point can be unsafe if such a case happens but if conditions are supposed to work now as they are mutually exclusive and currently we do not have the code that calls method `getExecutable` with "PATH" argument so 2nd condition is always performed, but definitely to be safe for the future this check should be present. Bug in 2nd point could also potentially occur but only if the case described in 1st point occurs at the same time with it, because entrySet()'s iterator traverses the elements (map entries) in ascending key order so PATH will never be 1st in row unless there no other elements before it in entry set so there should be no problem now if "PATH" is not passed as method parameter, but if it is passed then check in previous case will also help.

So we have add the check if "PATH" is passed as parameter inside first condition (without changing the order of conditions) to eliminate assigning the value of path to home variable. I have added the change. Please check now.